File: app/src/main/java/com/sorcerer/sorcery/iconpack/ui/others/ViewFader.java
package com.sorcerer.sorcery.iconpack.ui.others;

import android.view.View;
import android.view.ViewGroup;

/**
 * @description:
 * @author: Sorcerer
 * @date: 2016/11/4
 */

public class ViewFader {
    public void hideContentOf(ViewGroup viewGroup) {
        for (int i = 0; i < viewGroup.getChildCount(); i++) {
            viewGroup.getChildAt(i).setVisibility(View.GONE);
        }
    }

    public void showContent(ViewGroup viewGroup){
        for(int i=0;i<viewGroup.getChildCount();i++){
            viewGroup.getChildAt(i).setVisibility(View.VISIBLE);
        }
    }
}

File: app/src/main/java/com/sorcerer/sorcery/iconpack/ui/anim/ViewFader.kt
package com.sorcerer.sorcery.iconpack.ui.anim

import android.view.View
import android.view.ViewGroup

/**
 * @description:
 * *
 * @author: Sorcerer
 * *
 * @date: 2016/11/4
 */

class ViewFader {
    fun hideContentOf(viewGroup: ViewGroup) {
        for (i in 0..viewGroup.childCount - 1) {
            viewGroup.getChildAt(i).visibility = View.GONE
        }
    }

    fun showContent(viewGroup: ViewGroup) {
        for (i in 0..viewGroup.childCount - 1) {
            viewGroup.getChildAt(i).visibility = View.VISIBLE
        }
    }
}
